File: app/src/main/java/com/bearcave/passageplanning/base/database/BaseTable.java
package com.bearcave.passageplanning.base.database;


import com.bearcave.passageplanning.base.database.withcustomkey.BaseTableWithCustomKey;
import com.bearcave.passageplanning.data.database.ManagerListener;

public abstract class BaseTable<Dao extends DatabaseElement> extends BaseTableWithCustomKey<Dao, Integer>
        implements ManagerListener, CRUD<Dao> {

    public BaseTable(ManagerListener manager) {
        super(manager);
    }

    @Override
    protected String getIdKey(){
        return KEY_ID;
    }
}

File: app/src/main/java/com/bearcave/passageplanning/base/database/BaseTable.kt
package com.bearcave.passageplanning.base.database


import com.bearcave.passageplanning.base.database.withcustomkey.BaseTableWithCustomKey
import com.bearcave.passageplanning.data.database.ManagerListener

abstract class BaseTable<Dao : DatabaseElement>(manager: ManagerListener)
    : BaseTableWithCustomKey<Dao, Int>(manager),
        ManagerListener,
        CRUD<Dao> {

    override val idKey: String
        get() = BaseTableWithCustomKey.KEY_ID
}